dotnet/generate-vs-workload.csharp

```diff
@@ -58,7 +64,7 @@ using (TextWriter writer = new StreamWriter (outputPath)) {
 	writer.WriteLine ($"<Project>");
 	writer.WriteLine ($"  <PropertyGroup>");
 	var allPlatforms = string.Join (".", platforms.Select (v => v.Item1).OrderBy (v => v));
-	writer.WriteLine ($"    <TargetName>Microsoft.NET.Sdk.{allPlatforms}.Workload.{tfm}</TargetName>");
+	writer.WriteLine ($"    <TargetName>Microsoft.NET.Sdk.{allPlatforms}.Workload.{tfm}{branchName}</TargetName>");
```

Comment:
@rolfbjarne Is my assumption correct that we only plan on inserting/shipping content from release/8.0.1xx-xcode15.0 as a dependency of release/9.0.1xx-preview* or another branch?

Reply:
Yes, that's correct.

In addition to `release/9.0.1xx*` branches, this branch will also be a dependency of `release/8.0.1xx-xcode15.3` when we release that branch.

Reply:
One downside that I can see to this approach is that VS will need a manual update every time there is a new branch that carries a supplemental dependency, as that will cause the manifest file name to change. I'm wondering if it would make sense to try to use some more generic name here...

Will it be likely for release/9.0.1xx* or other branches to take dependencies on more than two branches? What determines when we create this type of "multitargeting support dependency" branch? Do we have a policy / upper limit where only latest and +1 is supported or can there be many dependencies?

Reply:
The idea is to support the earliest + latest OS version. Then add the fact that we typically support the previous + the current .NET version, and we end up with a total of 4 versions multi targeting supports (1 current + 3 older ones).

Tentative example for around Christmas 2024, with iOS 17.4 being the latest iOS 17 release, and iOS 18.2 being the latest iOS 18 release:

* net8.0-ios17.0
* net8.0-ios17.4 (latest iOS 17 release)
* net9.0-ios18.0
* net9.0-ios18.2 (latest iOS 18 release).

Then we might also add preview releases:

* net9.0-ios19.0 (during iOS 19 beta period / after WWDC 2025).

We might also (even though we don't really want to) end up adding more supported versions if we determine that the maintenance cost is worth it.

Reply:
Ok it sounds like we definitely want a relatively unique name for each manifest in that case, using the branch name as is done here should work fine for the cases you've mentioned. The downside is that there is more manual insertion work for us, but we can invest in more release tooling updates if that ends up being a major headache.